### PR TITLE
Add some additional logging to client

### PR DESCRIPTION
**Pull request by marcelveldt**

### CODE REVIEW

File: matter_server/client/client.py [hunk 1]
```diff
@@ -407,29 +406,40 @@ def _handle_event_message(self, msg: EventMessage) -> None:
                 event = EventType.NODE_ADDED
                 node = MatterNode(node_data)
                 self._nodes[node.node_id] = node
+                self.logger.debug("New node added: %s", node.node_id)
             else:
                 event = EventType.NODE_UPDATED
                 node.update(node_data)
+                self.logger.debug("Node updated: %s", node.node_id)
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply matter_server/client/client.py:413]: this one is not resource intensive so should we ?

File: matter_server/client/client.py [hunk 2]
```diff
@@ -407,29 +406,40 @@ def _handle_event_message(self, msg: EventMessage) -> None:
...
             )
             # cleanup endpoint only after signalling subscribers
             if node := self._nodes.get(node_id):
-                node.endpoints.pop(msg.data["endpoint_id"], None)
+                node.endpoints.pop(endpoint_id, None)
             return
         if msg.event == EventType.ATTRIBUTE_UPDATED:
             # data is tuple[node_id, attribute_path, new_value]
             node_id, attribute_path, new_value = msg.data
+            self.logger.debug(
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply matter_server/client/client.py:437]: strictly taken nothing intensive is done here so not sure about the benefit of guarding it

File: matter_server/client/client.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MartinHjelmare) [reply matter_server/client/client.py:413]: What events are used when the a device updates its state, ie events that lead to entity state updates in Home Assistant? Aren't those happening potentially a lot?

[the remark this answers is not in the document]
Comment (marcelveldt) [reply matter_server/client/client.py:413]: attribute updated events and yes that may happen a lot (but not that much) - say like somewhere between once a second and once every couple of minutes ?

**Review by MartinHjelmare — APPROVED** [review]

File: matter_server/client/client.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (marcelveldt) [reply matter_server/client/client.py:413]: I added the guard for the attribute changed callback anyway, just to be safe.

**Pull request merged**